### ISSUE TITLE
CWnd page takes forever to load!

### ISSUE THREAD

**Issue by qtormanen** [MicrosoftDocs/cpp-docs#4680]
The https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170 page now takes 10s of seconds to load. It didn't used to be this way. Can this please get fixed? It's a very common page to need to reference in MFC.

**Comment by orcmid** [comment]
There are 418 topics on that page.  It is ginormous.  I had to instruct the Edge browser to keep waiting several times.  

It also takes a while before the sidebar TOC shows up with the list of all MFC classes.  Probably the only solution for delivery over the web is smaller pages.

@qtormanen For off-line work, you could get/recommend the PDF except that's 9,342 pages 9IN 143MB) covering the complete MFC.  Unfortunately, it also takes several seconds to scroll down the table of contents to CWnd :(.

**Comment by qtormanen** [comment]
Yeah, I knew it was huge, but I wish Microsoft made the topic usable since it’s pretty foundational to MFC. Maybe it has always been this way since the topic went online (vs. the locally-installed MSDN library of a decade or two ago).

Thanks for the tip on the PDF. It’s nice to know it’s there.

--Quinton

**Comment by JoshHarmon** [comment]
> Maybe it has always been this way since the topic went online (vs. the locally-installed MSDN library of a decade or two ago).

No. It broke in 2023.

> Probably the only solution for delivery over the web is smaller pages.

Or whoever wrote javascript that makes a bazillion `insertAdjacentElement` calls and consequently causes container-ship-loads of style recalculations could write code that... doesn't do that. (See https://github.com/MicrosoftDocs/cpp-docs/issues/4593)

I took a look at the page via the internet archive and performance was fine on 10/25/22 through 1/9/2023, in that sure the page took a moment to load due to being pulled from the archive, but afterwards the TOC loaded performantly compared to today and never came even close to popping a Wait dialog.

Perf seemed to start perhaps regressing in the 3/28/2023 copy, and seemed to have unquestionably regressed by the 6/26/2023 copy (note that #4593 was originally filed 6/19/2023, tightening the bound slightly more). I counted from the main content such as the heading starting to render (FCP?) to avoid as much influence as possible from IA infra slowness, and when in question, I ran a couple of trials to see what variability, if any, existed in the post-FCP performance or whether a Wait dialog would pop reliably.

| Date | Link | Perf OK? |
|--------|--------|--------|
| 10/25/22 | [link](https://web.archive.org/web/20221025053530/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| OK|
| 12/5/22 | [link](https://web.archive.org/web/20221205043821/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| OK |
| 12/22/22 | [link](https://web.archive.org/web/20221222031558/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| OK |
| 1/9/23 | [link](https://web.archive.org/web/20230109155921/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| OK |
| 3/28/23 | [link](https://web.archive.org/web/20230328035232/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| mostly NO | 
| 6/26/23 | [link](https://web.archive.org/web/20230626082656/https://learn.microsoft.com/en-us/cpp/mfc/reference/cwnd-class?view=msvc-170)| NO |

**Comment by orcmid** [comment]
@qtormanen A workaround may be to view the `CWnd` page directly on GitHub.

It is at
[MSDocs cpp MFC reference Cwnd class](https://github.com/MicrosoftDocs/cpp-docs/blob/main/docs/mfc/reference/cwnd-class.md).

Selecting preview causes a web-rendering to appear quickly and you can scroll through it.   In the Edge browser, you can right-click on a previewed page and select print.  It has 348 pages, so switching to a PDF printer might be best.  And/Or use the print dialog to reduce printing to a selected group of pages.

The markdown page has not been touched in 7 months, so this might be good enough.

**Comment by TylerMSFT** [comment]
Moved this issue to an internal work item tracking system because we are moving away from using GitHub issues to track C++ doc work.
